lib/node/disseminator.ml

```diff
@@ -0,0 +1,53 @@
+type pool_elt = {
+  message : Message.t;
+  remaining : int;
+}
+
+module DigestSet = Set.Make (Digest)
+
+type t = {
+  round : int;
+  pool : pool_elt list;
+  num_rounds : int;
+  epoch_length : float;
+  seen : DigestSet.t;
+}
+
+let create num_rounds epoch_length =
+  { round = 0; pool = []; num_rounds; epoch_length; seen = DigestSet.empty }
+
+let next_round disseminator =
+  let time = Unix.time () in
```

Comment:
`Unix.time` is in seconds, is it precise enough?

Reply:
I think seconds is precise enough yes, this is just a matter of whether we think a message is too old to be "hot" anymore